src/commands/image/imageSource/ContainerAppStartVerificationStep.ts

```diff
@@ -0,0 +1,200 @@
+
+        const resourceGroupName: string = nonNullValueAndProp(context.containerApp, 'resourceGroup');
+        const containerAppName: string = nonNullValueAndProp(context.containerApp, 'name');
+
+        let revision: Revision | undefined;
+        let revisions: Revision[];
+
+        const start: number = Date.now();
+
+        while (true) {
+            if ((Date.now() - start) > maxWaitTimeMs) {
+                break;
+            }
+
+            await delay(1000);
+
+            revisions = await uiUtils.listAllIterator(this._client.containerAppsRevisions.listRevisions(resourceGroupName, containerAppName));
+            revision = revisions.find(r => r.name === context.containerApp?.latestRevisionName && r.template?.containers?.[context.containersIdx ?? 0].image === context.image);
+
+            if (revision) {
+                return revision.id;
+            }
+        }
+
+        return undefined;
+    }
+
+    private async waitAndGetRevisionStatus(context: T, revisionId: string, containerAppName: string, maxWaitTimeMs: number): Promise<string | undefined> {
+        this._client ??= await createContainerAppsAPIClient([context, createSubscriptionContext(context.subscription)]);
+        const parsedRevision = parseAzureResourceId(revisionId);
+
+        let revision: Revision;
+        const start: number = Date.now();
+
+        while (true) {
+            if ((Date.now() - start) > maxWaitTimeMs) {
+                break;
+            }
+
+            await delay(2000);
+
+            revision = await this._client.containerAppsRevisions.getRevision(parsedRevision.resourceGroup, containerAppName, parsedRevision.resourceName);
+
+            if (
+                revision.provisioningState === KnownRevisionProvisioningState.Deprovisioning ||
+                revision.provisioningState === KnownRevisionProvisioningState.Provisioning ||
+                revision.runningState === KnownRevisionRunningState.Processing ||
+                revision.runningState === 'Activating' // For some reason this isn't listed in the known enum
+            ) {
+                continue;
+            }
+
+            return revision.runningState;
+        }
+
+        return undefined;
+    }
+
+    /**
+     * Try to query for any logs associated with the revision and add them to the Copilot activity attributes
+     */
+    private async tryAddLogAttributes(context: T, revisionName: string) {
+        // Basic validation check since we're including a name directly in the query
+        if (revisionName.length > 54 || !/^[\w-]+$/.test(revisionName)) {
+            const invalidName: string = localize('unexpectedRevisionName', 'Internal warning: Encountered an unexpected revision name format "{0}". Skipping log query for the revision status check.', revisionName);
+            ext.outputChannel.appendLog(invalidName);
+            throw new Error(invalidName);
```

Comment:
Should this actually throw an error? Keeping in mind that if this step fails, it'll make the whole deployment be considered a failed operation.

Reply:
The error is caught, logged into telemetry, and then swallowed, so it should be fine!